core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,25 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Generic test that first adds a "boost" field to the otherwise randomly created query and
+     * then replaces the field with an unknown field. This should trigger ParsingException that
+     * points out the unknown field.
+     * Queries that allow arbitrary field names at this level need to override this test so it
+     * doesn't get executed for those specific tests.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        testQuery.boost(1.0f);
+        String queryAsString = testQuery.toString().replace("boost", "bogusField");
```

Comment:
Thats a good idea, although e.g. bool has a high chance of having a subclause with "boost" in it, it's likely that we skip the test then most of the times.